scripts/zones/RuAun_Gardens/mobs/Seiryu.lua

```diff
@@ -8,33 +8,49 @@ mixins = { require('scripts/mixins/job_special') }
 ---@type TMobEntity
 local entity = {}
 
-entity.onMobSpawn = function(mob)
-    mob:setMobMod(xi.mobMod.CANNOT_GUARD, 1)
-    GetNPCByID(ID.npc.PORTAL_OFFSET + 2):setAnimation(xi.anim.CLOSE_DOOR)
-end
-
 entity.onMobInitialize = function(mob)
     mob:setMobMod(xi.mobMod.ADD_EFFECT, 1)
     mob:setMobMod(xi.mobMod.IDLE_DESPAWN, 300)
-    mob:setMobMod(xi.mobMod.GIL_MIN, 1500)
-    mob:setMobMod(xi.mobMod.GIL_MAX, 30000)
+    mob:setMobMod(xi.mobMod.GIL_MIN, 18000)
+    mob:setMobMod(xi.mobMod.GIL_MAX, 18000)
 end
 
-entity.onMobMagicPrepare = function(mob, target, spellId)
-    if not mob:hasStatusEffect(xi.effect.HUNDRED_FISTS, 0) then
-        local rnd = math.random(1, 100)
-        if rnd <= 50 then
-            return 186 -- aeroga 3
-        elseif rnd <= 70 then
-            return 157 -- aero 4
-        elseif rnd <= 90 then
-            return 208 -- tornado
-        else
-            return 237 -- choke
+entity.onMobSpawn = function(mob)
+    mob:messageText(mob, ID.text.SKY_GOD_OFFSET + 9) -- Spawn message
+    GetNPCByID(ID.npc.PORTAL_OFFSET + 2):setAnimation(xi.anim.CLOSE_DOOR)
+    mob:setMobMod(xi.mobMod.CANNOT_GUARD, 1)
+    mob:setMobMod(xi.mobMod.WEAPON_BONUS, 34)
+    mob:setMobMod(xi.mobMod.MAGIC_COOL, 35)
+    mob:setMod(xi.mod.REGAIN, 450) -- Uses TP move every 20 seconds
+
+    -- Sky gods wait 10 seconds after spawning to start casting
+    mob:setMagicCastingEnabled(false)
+    mob:timer(10000, function(mobArg)
+        if mobArg then
+            mobArg:setMagicCastingEnabled(true)
+        end
+    end)
+
+    mob:addListener('EFFECT_LOSE', 'SEIRYU_HF', function(mobArg, effect)
+        if effect:getEffectType() == xi.effect.HUNDRED_FISTS then
+            mobArg:setMagicCastingEnabled(true)
+            mobArg:setMobAbilityEnabled(true)
         end
+    end)
+end
+
+entity.onMobFight = function(mob, target)
+    -- Increases to TP move usage every 9 seconds post 50% health
+    if mob:getHPP() < 50 and mob:getMod(xi.mod.REGAIN) < 700 then
+        mob:setMod(xi.mod.REGAIN, 700)
     end
+end
```

Comment:
If seiyuu were to heal itself, like when idle, the regain should be reverted back, I assume?
Also, checking for the mod is unnecesary.
```lua
    if mob:getHPP() < 50 then
        mob:setMod(xi.mod.REGAIN, 700)
    else
        mob:setMod(xi.mod.REGAIN, 450)
    end